File: all/1784_check_if_binary_string_has_at_most_one_segment_of_ones/sol_better_logic.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        # As the starting character of s is 1, we just need to check if there
        # has '01' in s that will start second continuous segment of ones.
        return '01' not in s


if '__main__' == __name__:
    sol = Solution()
    testcases_list = [
        ['1001'],
        ['110'],
        ['1'],
    ]

    for testcase in testcases_list:
        print(sol.checkOnesSegment(*testcase))

File: all/1784_check_if_binary_string_has_at_most_one_segment_of_ones/sol_better_logic.c
#include <stdbool.h>

bool checkOnesSegment(char* s) {
    /*
     * As the starting character of s is 1, we just need to check if there has
     * '01' in s that will start second continuous segment of ones.
     */
    for (int i = 1; '\0' != s[i]; ++i) {
        if (('0' == s[i - 1]) && ('1' == s[i])) {
            return false;
        }
    }

    return true;
}
